system/Security/Security.php

```diff
@@ -528,13 +529,18 @@ private function saveHashInCookie(): void
                 'expires' => $this->expires === 0 ? 0 : time() + $this->expires,
             ]
         );
-        $this->sendCookie($this->request);
+
+        /** @var Response $response */
+        $response = Services::response();
+        $response->setCookie($this->cookie);
```

Comment:
Any point of storing response service in a variable?

```suggestion
        Services::response()ٍ->setCookie($this->cookie);
```

Reply:
To explicitly indicate dependency.
This is also to avoid disabling the deptrac check.

You may like the short code, but it hides the dependency Response.
deptrac can't detect it.
```php
Services::response()->setCookie($this->cookie);
```